x/sudo/sudo.go

```diff
@@ -0,0 +1,236 @@
+package sudo
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/NibiruChain/collections"
+	"github.com/cosmos/cosmos-sdk/codec"
+
+	"github.com/NibiruChain/nibiru/x/common/set"
+	"github.com/NibiruChain/nibiru/x/sudo/pb"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+type Keeper struct {
+	Sudoers collections.Item[PbSudoers]
+}
+
+func NewKeeper(
+	cdc codec.BinaryCodec,
+	storeKey sdk.StoreKey,
+) Keeper {
+	return Keeper{
+		Sudoers: collections.NewItem(storeKey, 1, SudoersValueEncoder(cdc)),
+	}
+}
+
+var ROOT_ACTIONS = pb.ROOT_ACTIONS
+var ROOT_ACTION = pb.ROOT_ACTION
+
+func NewHandler(k Keeper) sdk.Handler {
+	return func(ctx sdk.Context, msg sdk.Msg) (*sdk.Result, error) {
+		goCtx := sdk.WrapSDKContext(
+			ctx.WithEventManager(sdk.NewEventManager()),
+		)
+		switch msg := msg.(type) {
+		case *pb.MsgEditSudoers:
+			res, err := k.EditSudoers(goCtx, msg)
+			return sdk.WrapServiceResult(ctx, res, err)
+
+		default:
+			errMsg := fmt.Sprintf(
+				"unrecognized %s message type: %T", pb.ModuleName, msg)
+			return nil, sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, errMsg)
+		}
+	}
+}
+
+// Ensure the interface is properly implemented at compile time
+var _ pb.MsgServer = Keeper{}
+
+func (k Keeper) EditSudoers(
+	goCtx context.Context, msg *pb.MsgEditSudoers,
```

Comment:
I would validate the message outside of this function, and the default should return an error.

Reply:
Good idea, updated!